TrackletAlgorithm/ProjectionCalculator.h

```diff
@@ -37,43 +39,54 @@ PC::addProj(const TrackletProjection<TProjType> &proj, const BXType bx, Tracklet
   else
     phi >>= 3;
 
+  int ipage = trackletIndex >> 7;
+
+  trackletIndex = trackletIndex&127;
+
   if (NProjOut > 0 && TPROJMask & (0x1 << 0) && success && proj_success && phi == 0)
-    projout[0].write_mem(bx, proj);
+    projout[0].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 1 && TPROJMask & (0x1 << 1) && success && proj_success && phi == 1)
-    projout[1].write_mem(bx, proj);
+    projout[1].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 2 && TPROJMask & (0x1 << 2) && success && proj_success && phi == 2)
-    projout[2].write_mem(bx, proj);
+    projout[2].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 3 && TPROJMask & (0x1 << 3) && success && proj_success && phi == 3)
-    projout[3].write_mem(bx, proj);
+    projout[3].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 4 && TPROJMask & (0x1 << 4) && success && proj_success && phi == 4)
-    projout[4].write_mem(bx, proj);
+    projout[4].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 5 && TPROJMask & (0x1 << 5) && success && proj_success && phi == 5)
-    projout[5].write_mem(bx, proj);
+    projout[5].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 6 && TPROJMask & (0x1 << 6) && success && proj_success && phi == 6)
-    projout[6].write_mem(bx, proj);
+    projout[6].write_mem(bx, proj, trackletIndex, ipage);
   if (NProjOut > 7 && TPROJMask & (0x1 << 7) && success && proj_success && phi == 7)
-    projout[7].write_mem(bx, proj);
+    projout[7].write_mem(bx, proj, trackletIndex, ipage);
 
   return (success && proj_success);
 }
 
 // This is the primary interface for the ProjectionCalculator.
 template<
   TF::seed Seed, // seed layer combination (TC::L1L2, TC::L3L4, etc.)
-  TC::itc iTC    // letter at the end of the TC name (TC_L1L2A and TC_L5L6A have
+  TC::itc iTC,    // letter at the end of the TC name (TC_L1L2A and TC_L5L6A have
+  unsigned int MaskBarrel,
+  unsigned int MaskDisk
   > void
   ProjectionCalculator(
     const BXType bx,
     BXType& bx_o,
     TrackletParameters const &tpar, int trackletIndex,
```

Comment:
@aryd Should `trackletIndex` be defined as an `ap_uint<kNBits_MemPage+kNBits_MemAddr>` (i.e., `ap_uint<9>`), to match the argument to `addProj`?
```suggestion
    TrackletParameters const &tpar, ap_uint<kNBits_MemPage+kNBits_MemAddr> trackletIndex,
```

Reply:
Yes - fixed. (This should be reduced to only two bits, but will require some additional changes that I plan to do later.)